packages/edit-site/src/components/sidebar-navigation-screen-page/index.js

```diff
@@ -36,127 +32,34 @@ import { unlock } from '../../private-apis';
-
-function getPageDetails( page ) {
-	if ( ! page ) {
-		return [];
-	}
-
-	const details = [
-		{
-			label: __( 'Status' ),
-			value: (
-				<StatusLabel
-					status={ page?.password ? 'protected' : page.status }
-					date={ page?.date }
-				/>
-			),
-		},
-		{
-			label: __( 'Slug' ),
-			value: <Truncate numberOfLines={ 1 }>{ page.slug }</Truncate>,
-		},
-	];
-
-	if ( page?.templateTitle ) {
-		details.push( {
-			label: __( 'Template' ),
-			value: page.templateTitle,
-		} );
-	}
-
-	details.push( {
-		label: __( 'Parent' ),
-		value: page?.parentTitle,
-	} );
-
-	/*
-	 * translators: If your word count is based on single characters (e.g. East Asian characters),
-	 * enter 'characters_excluding_spaces' or 'characters_including_spaces'. Otherwise, enter 'words'.
-	 * Do not translate into your own language.
-	 */
-	const wordCountType = _x( 'words', 'Word count type. Do not translate!' );
-	const wordsCounted = page?.content?.rendered
-		? wordCount( page.content.rendered, wordCountType )
-		: 0;
-	const readingTime = Math.round( wordsCounted / AVERAGE_READING_RATE );
-
-	if ( wordsCounted ) {
-		details.push(
-			{
-				label: __( 'Words' ),
-				value: wordsCounted.toLocaleString() || __( 'Unknown' ),
-			},
-			{
-				label: __( 'Time to read' ),
-				value:
-					readingTime > 1
-						? sprintf(
-								/* translators: %s: is the number of minutes. */
-								__( '%s mins' ),
-								readingTime.toLocaleString()
-						  )
-						: __( '< 1 min' ),
-			}
-		);
-	}
-	return details;
-}
+import PageDetails from './page-details';
 
 export default function SidebarNavigationScreenPage() {
 	const { setCanvasMode } = unlock( useDispatch( editSiteStore ) );
-	const { getFeaturedMediaDetails } = unlock( privateEditorApis );
 	const {
 		params: { postId },
 	} = useNavigator();
 	const { record } = useEntityRecord( 'postType', 'page', postId );
 
 	const {
-		parentTitle,
-		featuredMediaDetails: { mediaSourceUrl, altText },
-		templateSlug,
+		featuredMediaDetails: { sourceUrl, altText },
 	} = useSelect(
 		( select ) => {
-			const { getEditedPostContext } = unlock( select( editSiteStore ) );
-
+			const { getEntityRecord } = select( coreStore );
 			// Featured image.
 			const attachedMedia = record?.featured_media
-				? select( coreStore ).getEntityRecord(
+				? getEntityRecord(
 						'postType',
 						'attachment',
 						record?.featured_media
 				  )
 				: null;
 
-			// Parent page title.
-			let _parentTitle = record?.parent
-				? select( coreStore ).getEntityRecord(
-						'postType',
-						'page',
-						record.parent,
-						{ _fields: [ 'title' ] }
-				  )
-				: null;
-
-			if ( _parentTitle?.title ) {
-				_parentTitle = decodeEntities(
-					_parentTitle.title?.rendered || __( '(no title)' )
-				);
-			} else {
-				_parentTitle = __( 'Top level' );
-			}
-
 			return {
-				parentTitle: _parentTitle,
-				templateSlug: getEditedPostContext()?.templateSlug,
 				featuredMediaDetails: {
```

Comment:
Thanks for the PR, this is the most important thing to fix IMO, let's avoid nested objects here.

Reply:
Dang, I had the commit ready, but didn't push late last night. Thanks for the nudge. ✅